src/generate-elements.ts

```diff
@@ -237,3 +237,21 @@ class AnalysisWalker {
     }
   }
 }
+
+/**
+ * Implements Polymer core's translation of property names to attribute names.
+ *
+ * Returns null if the property name cannot be so converted.
+ */
+function propertyToAttributeName(propertyName: string): string|null {
+  // Polymer core will not map a property name that starts with an uppercase
+  // character onto an attribute.
+  if (propertyName[0].toUpperCase() === propertyName[0]) {
+    return null;
+  }
+  return propertyName
+      .replace(
+          /(.)([A-Z])/g,
+          (_: string, c1: string, c2: string) => `${c1}-${c2.toLowerCase()}`)
+      .toLowerCase();
```

Comment:
Done, also simplified the implementation (and fixed a bug with consecutive caps that I missed earlier, and mis-tested)